src/cpu/aarch64/jit_uni_eltwise.cpp

```diff
@@ -421,49 +424,72 @@ inline void jit_uni_kernel_t<cpu_isa_t::sve_128>::pack_fp16(
 
-    bool ok = mayiuse(isa) && is_fwd()
-            && utils::everyone_is(
-                    d_type, src_md()->data_type, dst_md()->data_type)
-            && !has_zero_dim_memory() && src_d.is_dense(true)
-            && eltwise_injector::is_supported(isa, desc_.alg_kind)
-            // refer to a comment in jit_uni_kernel why this is needed
-            && IMPLICATION(!src_d.is_dense(), is_zero_preserved())
-            && attr()->has_default_values() && set_default_formats_common()
-            && src_d == memory_desc_wrapper(dst_md());
-    return ok ? status::success : status::unimplemented;
+    VDISPATCH_ELTWISE(mayiuse(isa), VERBOSE_UNSUPPORTED_ISA);
+    VDISPATCH_ELTWISE(is_fwd(), VERBOSE_BAD_PROPKIND);
+
+    VDISPATCH_ELTWISE((src_md()->data_type == dst_md()->data_type),
+            VERBOSE_INCONSISTENT_DT, "src", "dst");
+    VDISPATCH_ELTWISE(utils::one_of(src_md()->data_type, data_type::f32,
+                              data_type::bf16, data_type::f16),
+            VERBOSE_UNSUPPORTED_DT);
+
+    VDISPATCH_ELTWISE(!has_zero_dim_memory(), VERBOSE_EMPTY_TENSOR, "data");
+    VDISPATCH_ELTWISE(src_d.is_dense(true), VERBOSE_NONTRIVIAL_STRIDE);
+
+    VDISPATCH_ELTWISE(eltwise_injector::is_supported(isa, desc_.alg_kind),
+            VERBOSE_BAD_ALGORITHM);
+    // refer to a comment in jit_uni_kernel why this is needed
+    VDISPATCH_ELTWISE(IMPLICATION(!src_d.is_dense(), is_zero_preserved()),
+            VERBOSE_UNSUPPORTED_ATTR);
+    VDISPATCH_ELTWISE(attr()->has_default_values(), VERBOSE_UNSUPPORTED_ATTR);
+    VDISPATCH_ELTWISE(set_default_formats_common(), VERBOSE_UNSUPPORTED_TAG);
+    VDISPATCH_ELTWISE(src_d == memory_desc_wrapper(dst_md()),
+            VERBOSE_INCONSISTENT_MDS, "src", "dst");
+
+    return status::success;
 }
 
-template <cpu_isa_t isa, data_type_t d_type>
-jit_uni_eltwise_fwd_t<isa, d_type>::jit_uni_eltwise_fwd_t(const pd_t *apd)
+template <cpu_isa_t isa>
+jit_uni_eltwise_fwd_t<isa>::jit_uni_eltwise_fwd_t(const pd_t *apd)
     : primitive_t(apd) {}
 
-template <cpu_isa_t isa, data_type_t d_type>
-jit_uni_eltwise_fwd_t<isa, d_type>::~jit_uni_eltwise_fwd_t() = default;
+template <cpu_isa_t isa>
+jit_uni_eltwise_fwd_t<isa>::~jit_uni_eltwise_fwd_t() = default;
 
-template <cpu_isa_t isa, data_type_t d_type>
-status_t jit_uni_eltwise_fwd_t<isa, d_type>::init(engine_t *engine) {
+template <cpu_isa_t isa>
+status_t jit_uni_eltwise_fwd_t<isa>::init(engine_t *engine) {
     CHECK(safe_ptr_assign(kernel_, new jit_uni_kernel_t<isa>(pd())));
     return kernel_->create_kernel();
 }
 
-template <cpu_isa_t isa, data_type_t d_type>
-status_t jit_uni_eltwise_fwd_t<isa, d_type>::execute(
-        const exec_ctx_t &ctx) const {
-    auto src = CTX_IN_MEM(const data_t *, DNNL_ARG_SRC);
-    auto dst = CTX_OUT_MEM(data_t *, DNNL_ARG_DST);
+template <cpu_isa_t isa>
+status_t jit_uni_eltwise_fwd_t<isa>::execute(const exec_ctx_t &ctx) const {
+
+    const data_type_t src_dt = pd()->src_md()->data_type;
```

Comment:
nit: readability: I'd probably move this further down, closer to where you first actually make use of it.